### DISCUSSION TITLE
How do I get Flask-Babel to register Flask-Security messages for translation?

### DISCUSSION THREAD

**Discussion by magma764** [pallets-eco/flask-security#445] [Q&A] [answered]
I have referred to the documentation [here](https://flask-security-too.readthedocs.io/en/stable/customizing.html?highlight=translation#localization) and it is not clear to me. Do I need to change anything in my configuration?

My babel.cfg is very similar to the documented example:
```[python: **.py]
[jinja2: templates/security/**.html]
[jinja2: **/templates/**.html]
extensions=jinja2.ext.autoescape,jinja2.ext.with_
```
Upon running `pybabel extract -F babel.cfg -o messages.pot .` the file will generate an contain all my defined strings, but nothing from Flask-Security (flashed messages, form validation errors etc).

I have tried setting my domain to security, to no effect:
`babel = Babel(app, default_domain='security')`

What am I missing here?
Edit: I've been thinking that I could include the imported Flask-Security module location in `babel.cfg` - but this will differ on other machines, surely there is some other way?

**Comment by jwag956** [comment]
I get confused with babel as well.. So tell me a bit more about your directory structure - did you copy ALL the templates/security/** into your project? I am sure pybabel isnt going to look in site-packages!
Also - are you trying to change/add to shipped flask-security messages, or just use the shipped translations?

I assume you are using Flask-Security-Too 4.0.0 - in that case the templates actually reference Flask-Security translations directly and should work even if you copied and modified the template.

And don't set your domain to 'security'...

Out of the box, if you use flask-security templates (not copied) then translations should work - might be worth quickly making sure that is the case.

**Reply by magma764** [reply]
Hi jwag
I copied *most* of the templates from `templates/security` as I won't be using two factor, US etc.
I am using 4.0.0

> are you trying to change/add to shipped flask-security messages, or just use the shipped translations?

I need to add support for the obscure language which is Afrikaans, no need to change the English versions.
I think I should set it up like this in my app configuration file:
```python
app.config[SECURITY_MSG_ALREADY_CONFIRMED] = (_('Your email has already been confirmed'), 'info')
```

**Comment by jwag956** [comment, marked as answer]
Hmm - I think that is way too much work - as you have seen - translations come from code as well as templates.

One thought would be to simply fork flask-security-too - add a new translations directory for your LANG type - then run
python setup.py update_catalog - that should create a .po file you can modify to add the translations.

I haven't played with it - but it is possible you could simply copy an existing .po file and modify it. BABEL_TRANSLATIONS_DIRECTORIES can be set to point to your translations directory I believe...

Sorry - all a bit vague - hopefully give you some ideas.

You are more than welcome to submit a PR with your translations (as many as you do).
